src/utils/misc.ts

```diff
@@ -0,0 +1,14 @@
+type VERSIONS = "v2.0.0"
+
+export function deprecated(
+    thing: string,
+    version: VERSIONS,
+    opts?: { aggressive?: boolean; replacement?: string }
+): void {
+    let notice = `LZString | ${thing} is deprecated as of: ${version}`
+    if (opts?.replacement) {
+        notice += ` - Please use ${opts.replacement} instead`
+    }
+    console.error(notice)
+    if (window && opts?.aggressive) alert(notice)
```

Comment:
This isn't built for the browser, so not sure exactly what to do here, but maybe adding a `declare var window` or similar - not sure that this is the best thing to do blocking everything as some people might update without checking etc

Reply:
Fair enough, it is simple to take it out. Wanted to cover all the bases and with all the benchmarking pages that have been done I thought browser support was needed.

Reply:
It's needed - but TS / vite doesn't know it's there - so need to make sure that this doesn't throw an error when it doesn't exist :-)

Reply:
If you like, we can add "DOM" back to the lib compilerOptions in tsconfig.json so ts/vite stops throwing compile errors.

Reply:
That gives too much risk of something coming in that shouldn't - if the only place that exists is in this file then it feels safe - technically could disable the check on this line, but better to do it "properly" when it's a known :-)

Reply:
Removed browser-specific elements. We can always add the flags back later when we have a proper spec for browser support.